File: Gestor_canciones2/src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.List;

import gestorAplicacion.elementosLibreria.Cancion;
import gestorAplicacion.elementosLibreria.Biblioteca;
import gestorAplicacion.elementosLibreria.Favoritos;
import gestorAplicacion.elementosLibreria.Me_gusta;
import gestorAplicacion.elementosLibreria.Mis_artistas;
import gestorAplicacion.elementosLibreria.Playlist;
import gestorAplicacion.personas.Artista;
import gestorAplicacion.personas.Usuario;
/*
 * Se utiliza para serializar todos los objetos creados durante la ejecucion del proyecto
 * @author Erik Gonzalez
 * @author Felipe Miranda
 */

public class Serializador {
	/*
	 * Serializamos una lista por el nombre de la clase
	 * @param <E>       El generico se usa para poder agredar las clases que se crearon
	 * @param lista     Una lista de objetos
	 * @param className El nombre de la clase que queremos usar como nombre del archivo
	 */
	public static <E> void serializar(List<E> lista, String className) {
		FileOutputStream fileOut;

		try {
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			// Se crea un fileoutputstream para saber donde serializar los archivos
			fileOut = new FileOutputStream(path);
			// Se crea un objeto output stream para poder escribir en el archivo
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			// Guardamos la lista de objetos
			out.writeObject(lista);
			out.close();
			fileOut.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

	/* Serializamos todas las clases que necesitamos  */
	public static void serializarTodo() {
		Serializador.serializar(Biblioteca.getBibliotecas() , "Biblioteca");
		Serializador.serializar(Cancion.getCancionesCreadas(), "Canciones");
		Serializador.serializar(Favoritos.getFavoritos() , "Favotitos");
		Serializador.serializar(Me_gusta.getMe_gustas() , "Me gusta");
		Serializador.serializar(Mis_artistas.getMis_artistas() , "Mis artistas");
		Serializador.serializar(Playlist.getPlaylists(), "Playlists");
		Serializador.serializar(Artista.getArtistas(), "Artistas");
		Serializador.serializar(Usuario.getUsuarios(), "Usuarios");
	}

}

File: Phyton/Gestor_canciones/src/baseDatos/Serializador.py
import pickle
from gestorAplicacion.elementosLibreria.Cancion import Cancion
from gestorAplicacion.elementosLibreria.Biblioteca import Biblioteca
from gestorAplicacion.elementosLibreria.Favoritos import Favoritos
from gestorAplicacion.elementosLibreria.Me_gusta import Me_gusta
from gestorAplicacion.elementosLibreria.Mis_artistas import Mis_artistas
from gestorAplicacion.elementosLibreria.Playlist import Playlist
from gestorAplicacion.personas.Artista import Artista
from gestorAplicacion.personas.Usuario import Usuario
import pathlib
import os

def serializar(lista, className):
        def camino(className):
            string = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\"+className+".txt")
            return string
        try:
            # Creo el archivo pickle para guardar los objetos
            picklefile = open(camino(className), 'wb')
            # pickle el objeto en el archivo
            pickle.dump(lista, picklefile)
            # cierro el archivo para guardar
            picklefile.close()
            
        except:
            print("paila tuqui tuqui muñeco")

def serializarTodo():

        serializar(Biblioteca.bibliotecas, "Biblioteca")
        serializar(Cancion.cancionesCreadas, "Canciones")
        serializar(Favoritos.favoritos, "Favotitos")
        serializar(Me_gusta.me_gustas, "Me gusta")
        serializar(Mis_artistas.mis_artistas, "Mis artistas")
        serializar(Playlist.playlists, "Playlists")
        serializar(Artista.artistas, "Artistas")
        serializar(Usuario.usuarios, "Usuarios")

